controllers/module_reconciler_test.go

```diff
@@ -276,16 +275,28 @@ var _ = Describe("ModuleReconciler", func() {
 				},
 			}
 
-			c := fake.
-				NewClientBuilder().
-				WithScheme(scheme).
-				WithObjects(&mod, &ds).
-				WithLists(&nodeList).
-				Build()
+			ctx := context.TODO()
 
-			mr := controllers.NewModuleReconciler(c, mockBM, mockDC, mockKM, mockCU)
+			client := client.NewMockClient(ctrl)
+			gomock.InOrder(
+				client.EXPECT().Get(ctx, req.NamespacedName, gomock.Any()).DoAndReturn(
+					func(c context.Context, r types.NamespacedName, m *ootov1alpha1.Module) error {
+						m.ObjectMeta = mod.ObjectMeta
+						m.Spec = mod.Spec
+						return nil
+					},
+				),
+				client.EXPECT().List(ctx, gomock.Any(), gomock.Any()).DoAndReturn(
+					func(c context.Context, list *v1.NodeList, o interface{}) error {
+						list.Items = nodeList.Items
+						return nil
+					},
+				),
+				client.EXPECT().Get(ctx, gomock.Any(), gomock.Any()).Return(nil),
+				client.EXPECT().Patch(ctx, gomock.Any(), gomock.Any()).Return(nil),
```

Comment:
These two method calls were the most tricky to find because they are happening in the implementation of a hardcoded external dependency, instead of using an interface. The `Get()` and `Patch()` methods are used in the implementation of the 
`controllerutil.CreateOrPatch()` method.